src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +75,231 @@ const NotebookLauncher = _.flow(
+      const currentLocker = potentialLockers[lockedBy]
+      setLockedByEmail(currentLocker || null)
+    })
+    findLockedByEmail()
+  })
 
-const ReadOnlyMessage = ({ notebookName, workspace, workspace: { canCompute, workspace: { namespace, name } } }) => {
-  const [copying, setCopying] = useState(false)
+  return h(Modal, {
+    width: 530,
+    title: 'Notebook Is In Use',
+    onDismiss,
+    showButtons: false
+  }, [
+    p(lockedByEmail ? `This notebook is currently being edited by ${lockedByEmail}.` : `This notebook is currently locked because another user is editing it.`),
+    p('You can make a copy, or run it in Playground Mode to explore and execute its contents without saving any changes.'),
+    div({ style: { marginTop: '2rem' } }, [
+      h(ButtonSecondary, {
+        style: { paddingRight: '1rem', paddingLeft: '1rem' },
+        onClick: () => onDismiss()
+      }, ['CANCEL']),
+      h(ButtonSecondary, {
+        style: { paddingRight: '1rem', paddingLeft: '1rem' },
+        onClick: () => onCopy()
+      }, ['MAKE A COPY']),
+      h(ButtonPrimary, {
+        onClick: () => onPlayground()
+      }, ['RUN IN PLAYGROUND MODE'])
+    ])
+  ])
+}
 
-  return div({ style: { padding: '1rem 2rem', display: 'flex', alignItems: 'center' } }, [
-    div({ style: { fontSize: 16, fontWeight: 'bold', position: 'absolute' } },
-      ['Viewing read-only']),
+const PlaygroundModal = ({ onDismiss, onPlayground }) => {
+  const [hidePlaygroundMessage, setHidePlaygroundMessage] = useState(false)
+  return h(Modal, {
+    width: 530,
+    title: 'Playground Mode',
+    onDismiss,
+    okButton: h(ButtonPrimary,
+      {
+        onClick: () => {
+          BrowserStorage.setLocalPref('hidePlaygroundMessage', hidePlaygroundMessage.toString())
+          onPlayground()
+        }
+      },
+      'Continue')
+  }, [
+    `Playground mode allows you to explore, change, and run the code, but your edits will not be saved.`,
+    div({ style: { flexGrow: 1 } }),
+    'To save your work, choose Make a Copy from the File menu to make your own version.',
     div({ style: { flexGrow: 1 } }),
-    Utils.cond(
-      [!canCompute, () => h(ButtonOutline, { onClick: () => setCopying(true) }, ['copy to another workspace to edit'])],
-      () => h(ButtonOutline, {
-        href: Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
-      }, ['edit in Jupyter'])
+    h(LabeledCheckbox, {
+      checked: hidePlaygroundMessage,
+      onChange: v => setHidePlaygroundMessage(v)
+    }, [span({ style: { marginLeft: '0.5rem' } }, ['Do not show again '])])
+  ])
+}
+
+
+const PreviewHeader = ({ queryParams, cluster, readOnlyAccess, refreshClusters, notebookName, workspace, workspace: { canShare, workspace: { namespace, name, bucketName } } }) => {
+  const signal = useCancellation()
+  const { profile: { email } } = Utils.useAtom(authStore)
+  const hidePlaygroundModal = BrowserStorage.getLocalPref('hidePlaygroundMessage')
```

Comment:
Since this is reading a snapshot of a stateful value that might change, it would be better to avoid doing this in the render path to avoid possible subtle bugs. Instead, we can read it right as we're checking the value in the `onClick` below.